test/OpenFeature.IntegrationTests/Steps/EvaluationStepDefinitions.cs

```diff
@@ -0,0 +1,275 @@
+        public void Givenaproviderisregisteredwithcachedisabled()
+        {
+            //_scenarioContext.Pending();
+        }
+
+        [When(@"a boolean flag with key ""(.*)"" is evaluated with default value ""(.*)""")]
+        public void Whenabooleanflagwithkeyisevaluatedwithdefaultvalue(string flagKey, bool defaultValue)
+        {
+            this.booleanFlagValue = client.GetBooleanValue(flagKey, defaultValue);
+        }
+
+        [Then(@"the resolved boolean value should be ""(.*)""")]
+        public void Thentheresolvedbooleanvalueshouldbe(bool expectedValue)
+        {
+            Assert.Equal(expectedValue, this.booleanFlagValue.Result);
+        }
+
+        [When(@"a string flag with key ""(.*)"" is evaluated with default value ""(.*)""")]
+        public void Whenastringflagwithkeyisevaluatedwithdefaultvalue(string flagKey, string defaultValue)
+        {
+            this.stringFlagValue = client.GetStringValue(flagKey, defaultValue);
+        }
+
+        [Then(@"the resolved string value should be ""(.*)""")]
+        public void Thentheresolvedstringvalueshouldbe(string expected)
+        {
+            Assert.Equal(expected, this.stringFlagValue.Result);
+        }
+
+        [When(@"an integer flag with key ""(.*)"" is evaluated with default value (.*)")]
+        public void Whenanintegerflagwithkeyisevaluatedwithdefaultvalue(string flagKey, int defaultValue)
+        {
+            this.intFlagValue = client.GetIntegerValue(flagKey, defaultValue);
+        }
+
+        [Then(@"the resolved integer value should be (.*)")]
+        public void Thentheresolvedintegervalueshouldbe(int expected)
+        {
+            Assert.Equal(expected, this.intFlagValue.Result);
+        }
+
+        [When(@"a float flag with key ""(.*)"" is evaluated with default value (.*)")]
+        public void Whenafloatflagwithkeyisevaluatedwithdefaultvalue(string flagKey, double defaultValue)
+        {
+            this.doubleFlagValue = client.GetDoubleValue(flagKey, defaultValue);
+        }
+
+        [Then(@"the resolved float value should be (.*)")]
+        public void Thentheresolvedfloatvalueshouldbe(double expected)
+        {
+            Assert.Equal(expected, this.doubleFlagValue.Result);
+        }
+
+        [When(@"an object flag with key ""(.*)"" is evaluated with a null default value")]
+        public void Whenanobjectflagwithkeyisevaluatedwithanulldefaultvalue(string flagKey)
+        {
+            this.objectFlagValue = client.GetObjectValue(flagKey, new Value());
+        }
+
+        [Then(@"the resolved object value should be contain fields ""(.*)"", ""(.*)"", and ""(.*)"", with values ""(.*)"", ""(.*)"" and (.*), respectively")]
+        public void Thentheresolvedobjectvalueshouldbecontainfieldsandwithvaluesandrespectively(string boolField, string stringField, string numberField, bool boolValue, string stringValue, int numberValue)
+        {
+            Value value = this.objectFlagValue.Result;
```

Comment:
Not a huge deal, but `.Result` blocks the thread until the result returns. Using `await` attaches a hidden callback so other things keep executing. I honestly don't know how the test-runner works in this case, but the suite _may_ run faster if you switch all the `.Result`s with `await`s.

Because this isn't production code, it really doesn't matter too much though.

Reply:
We could try it, but the tests take like 2 seconds, so there isn't that much room for improvement